### PR TITLE
Add routes for user, and friend

### PR DESCRIPTION
**Pull request by AgostonSzepessy**
These commits add routes and database operations for User, and Friend. In short, it allows performing CRUD operations on both.

# Routes for User
* `GET /user`: retrieves user's data
* `POST /user`: updates user data excluding password
* `POST /user/password`: updates user password
* `DELETE /user`: deletes the user

# Routes for Friend
* `GET /friend`: retrieves all the friends a user has
* `POST /friend`: creates a new friend request
* `POST /friend/confirm`: confirms a friend request
* `DELETE /friend`: deletes a friend request

There are corresponding functions for these in `UserService`, `User`, `FriendService`, and `Friend`.

### CODE REVIEW

File: src/models/friend.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import { knex } from '../utils/knex';
+import { logger } from '../utils/logger';
+
+export class Friend {
+
+    /**
+     * Returns the list of friends that a user has
+     * @param username username of user
+     */
+    public static async getFriends(username: string) {
```
Comment (Stuart-Wilcox) [inline src/models/friend.ts:11]: Maybe this should return a list of User objects instead of a list of Friends

File: src/models/user.ts [hunk 2]
```diff
@@ -49,6 +49,49 @@ export class User {
...
+     */
+    public static async updateUser(username: string, email: string, fname: string, lname: string) {
+        const data = {
+            username,
+            fname,
+            email,
+            lname,
+        };
+
+        const password = await knex('User').update(data).where('User.username', '=', username).returning('password');
```
Comment (Stuart-Wilcox) [inline src/models/user.ts:67]: Why does the password need to be returned?

Comment (AgostonSzepessy) [reply src/models/user.ts:67]: We keep the password attached to the user object in other places, so I thought it'd be best to keep it consistent

File: src/models/friend.ts (hunk 1 above)
Comment (AgostonSzepessy) [reply src/models/friend.ts:11]: Yes that's a good idea

**Review by Stuart-Wilcox — APPROVED** [review]

**Pull request merged**